test/fake-intake/fakeintake/server.go

```diff
@@ -0,0 +1,153 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package fakeintake
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+)
+
+type payload struct {
+	timestamp time.Time
+	data      []byte
+}
+
+type FakeIntake struct {
+	mu     sync.RWMutex
+	server http.Server
+
+	payloadStore map[string][]payload
```

Comment:
I think we may want to have different handlers, and also worth separating the HTTP/Handler logic and the functional logic.

Reply:
I applied what discussed and agreed in the RFC, should we reconsider and open again for discussion ?

Reply:
It's only a matter of code structure and moving this in a `PayloadStore` to isolate HTTP serving and functional logic, there's no change in the processing code.

Reply:
Ah, I see what you mean. Was planning on taking care of storage as an injected dependency in a follow up PR - I have a task for that in my backlog.

Reply:
Can you clarify what you mean by 

> have different handlers

That is what made me think you wanted to have one handler per endpoint, while in the RFC we agreed on having no parsing logic in the fake intake, accepting any call to the intake. This allows supporting any endpoint, for free.

Reply:
Clarified: no need for different handlers. Will separate the logic in a follow up PR, addressed by AETT-72